crates/pico/tests/integration_tests.rs

```diff
@@ -0,0 +1,25 @@
+use common::owned::{evaluate_input, Input};
+use pico::database::Database;
+
+mod calc;
+mod common;
+
+#[test]
+fn returns_result() {
+    let mut db = Database::default();
+    let key = "test_expr";
+
+    let mut input = Input {
+        value: "2 + 2 * 2".to_string(),
+    };
+    input.set(&mut db, key);
+    let mut result = evaluate_input(&mut db, key);
+    assert_eq!(result, 6);
+
+    input = Input {
+        value: "(2 + 2) * 2".to_string(),
+    };
```

Comment:
can we add some tests that ensure that:

- if the overall expression hasn't changed, it isn't re-evaluated?
- if a subexpression hasn't changed, it isn't re-evaluated? e.g. if we have (1 + 2) * 3 and later calculate (1 + 2) * 4, we do not recalculate (1 + 2)?

This might require some restructuring of the calc to e.g. call a `evaluate_input` recursively

Reply:
I need to think how to expose this info. Currently we don't know how exactly we got the value outside of memo function